src/app/services/article.service.ts

```diff
@@ -1,25 +1,50 @@
-import { Injectable } from '@angular/core';
-import { Article } from "../interfaces/article";
-import { HttpClient } from "@angular/common/http";
-import { delay, switchMap, from, Observable, of, tap } from "rxjs";
-import { getHardcodedArticles, getHardcodedArticle } from './hardcoded.article';
+import {Injectable} from '@angular/core';
+import {Article} from "../interfaces/article";
+import {from, Observable, of, switchMap, take} from "rxjs";
+import {getHardcodedArticle} from './hardcoded.article';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ArticleService {
-  private articles: Article[] | null = [];
+  private articles: Article[] = [];
   private featuredArticles: string[] = [
-    "stories", 
-    "church-construction-complete", 
-    "is-peter-rock", 
+    "stories",
+    "church-construction-complete",
+    "is-peter-rock",
     "why-not-catholic"
   ];
   private defaultImageURI = "https://t4.ftcdn.net/jpg/02/97/78/03/360_F_297780357_pK8VCA7wctbTFusAGiCfcoxbJLRwC9Bs.jpg";
-  constructor(private http: HttpClient) {
+
+  constructor() {
   }
 
-  loadArticles = async () => {
+  public readOne(articleId: string | null): Observable<Article> {
+    return this.readAll().pipe(take(1),
+      switchMap(() => {
+        const article = this.articles.find(article => article.articleId === articleId);
+        return of(article || {
+          header: 'invalid-article',
+          body: 'invalid-article',
+          imageURI: 'none',
+          subCategory: 'Article',
+          articleId: 'invalid-article'
+        });
+      }));
+  }
+
+  public readAll(): Observable<Article[]> {
```

Comment:
I don't think we should load "ALL" articles all at once.

Reply:
We don't. We're using your implementation of _loadArticles()_ under the hood and caching the results.

Reply:
Then tha name should not be "readAll".

But we can go with it for now. We can change the names as the project evolves.